mono/mini/debugger-agent.c

```diff
@@ -4563,19 +4591,29 @@ ss_update (SingleStepReq *req, MonoJitInfo *ji, SeqPoint *sp, DebuggerTlsData *t
 	}
 
 	if (req->depth == STEP_DEPTH_INTO && req->size == STEP_SIZE_MIN && (sp->flags & MONO_SEQ_POINT_FLAG_NONEMPTY_STACK) && ss_req->start_method){
-		method = jinfo_get_method (ji);
 		ss_calculate_framecount (tls, ctx);
 		if (ss_req->start_method == method && req->nframes && tls->frame_count == req->nframes) {//Check also frame count(could be recursion)
 			DEBUG_PRINTF (1, "[%p] Seq point at nonempty stack %x while stepping in, continuing single stepping.\n", (gpointer) (gsize) mono_native_thread_id_get (), sp->il_offset);
 			return FALSE;
 		}
 	}
 
+	MonoDebugMethodAsyncInfo* asyncMethod = mono_debug_lookup_method_async_debug_info (method);
+	if (asyncMethod) {
```

Comment:
We add them in method-to-ir.c mostly because otherwise we can't place breakpoints(we don't have ManagedIL->naitve_offset relation AFAICT), but also so when we are stepping we step on top of yield_offset.

I was thinking about if I should cache `mono_debug_lookup_method_async_debug_info` calls... I figured `ss_start` isn't called often(imho should only be called when user request stepping)...